sda/cmd/api/api.go

```diff
@@ -317,39 +317,73 @@ func getFiles(c *gin.Context) {
 	c.JSON(200, files)
 }
 
+/*
+ingestFile handles requests to initiate ingestion of a file.
+This endpoint supports two input modes:
+1. By file ID (via the "fileid" query parameter): Looks up the user and file path from the database.
+2. By JSON payload: Expects a JSON body with user and file path.
+The function constructs an ingest message, validates it
+and sends it to the broker with the appropriate correlation ID.
+*/
 func ingestFile(c *gin.Context) {
-	var ingest schema.IngestionTrigger
-	if err := c.BindJSON(&ingest); err != nil {
-		c.AbortWithStatusJSON(
-			http.StatusBadRequest,
-			gin.H{
-				"error":  "json decoding : " + err.Error(),
-				"status": http.StatusBadRequest,
-			},
-		)
+	var (
+		ingest schema.IngestionTrigger
+		corrID string
+	)
```

Comment:
```suggestion
	ingest schema.IngestionTrigger
```

Only this should be needed.